src/intersection/first-solution/intersection.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Creates an array of values that are in both the first and the second arrays.
+ *
+ * @param {Array} [firstArr=[]] - First array to be compared for repeated values.
+ * @param {Array} [secondArr=[]] - Second array to be compared for repeated values.
+ * @returns {Array} Returns the new array with only intersected values.
+ */
+
+export const arrayIntersection = (firstArr = [], secondArr = []) => {
+  let newArrayIntersected = [];
+  // Test if the arguments are arrays and make sure that arguments are passed
+  if (!Array.isArray(firstArr) || !Array.isArray(secondArr)) {
+    throw new TypeError('The arguments must be arrays');
+  } else if (!firstArr.length || !secondArr.length) {
+    throw new Error('Must pas two arrays to intercept the values');
+  } else {
+    // test if arrays have the same values, return only the values matched
+    newArrayIntersected = [...new Set(firstArr.filter((x) => secondArr.includes(x)))];
```

Comment:
Why initialize the variable newArrayIntersected, since you never do an adding operation (push, shift etc..). Here you are creating a new Set and overriding the pointer. There is no way that this function will return the initialized array in the memory. This way it uses unnecessary heap space. 
https://medium.com/@ethannam/javascripts-memory-model-7c972cd2c239

Reply:
With this approach heap is used 3 times, 
1 line 10
2 new set
3 [...
Solution can use heap once with simple methods

Reply:
True, thanks for note this, there is no need of new variable, filter does not modify the array instead it returns a new array, but I keep the `else` statement because maintains readability  to the control structure.  Changes pushed